pkg/operator/k8sutil/endpoint.go

```diff
@@ -20,22 +20,22 @@ import (
 	"context"
 	"fmt"
 
-	v1 "k8s.io/api/core/v1"
+	discoveryv1 "k8s.io/api/discovery/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
 // CreateOrUpdateEndpoint creates a service or updates the service declaratively if it already exists.
-func CreateOrUpdateEndpoint(ctx context.Context, clientset kubernetes.Interface, namespace string, endpointDefinition *v1.Endpoints) (*v1.Endpoints, error) {
+func CreateOrUpdateEndpoint(ctx context.Context, clientset kubernetes.Interface, namespace string, endpointDefinition *discoveryv1.EndpointSlice) (*discoveryv1.EndpointSlice, error) {
 	name := endpointDefinition.Name
-	logger.Debugf("creating endpoint %q. %v", name, endpointDefinition.Subsets)
-	ep, err := clientset.CoreV1().Endpoints(namespace).Create(ctx, endpointDefinition, metav1.CreateOptions{})
+	logger.Debugf("creating endpoint %q. %v", name, endpointDefinition.Endpoints)
+	ep, err := clientset.DiscoveryV1().EndpointSlices(namespace).Create(ctx, endpointDefinition, metav1.CreateOptions{})
```

Comment:
Will the old `Endpoints` resource still be there if the user is upgrading the cluster?  If yes, then any plans to tackle that?

Reply:
Added code to delete the older `Endpoints` once the newer `EndpointSlice` resources are created.